table.cpp

```diff
@@ -12,10 +12,10 @@ using std::string;
 
 table_t::table_t(const string& conninfo, const string& name, const string& type, const columns_t& columns, const hstores_t& hstore_columns,
     const int srid, const int scale, const bool append, const bool slim, const bool drop_temp, const int hstore_mode,
-    const bool enable_hstore_index, const boost::optional<string>& table_space, const boost::optional<string>& table_space_index) :
+    const bool enable_hstore_index, const boost::optional<string>& table_space, const boost::optional<string>& table_space_index, const boost::optional<string>& schema) :
```

Comment:
https://github.com/openstreetmap/osm2pgsql/blob/b81f2d992cf96aebe84e24eb57cbf4c483a3df5a/output-pgsql.cpp#L710-L712 explains we've got this constructor.
```c++
//tremble in awe of this massive constructor! seriously we are trying to avoid passing an
//options object because we want to make use of the table_t in output_mutli_t which could
//have a different tablespace/hstores/etc per table
```
Without checking `git blame` I can tell that @kevinkreiser wrote that comment ;)

That's not to say it couldn't be refactored to avoid the massive constructor. Perhaps by putting table options into a `table_options_t` and passing that in?